redfish-core/lib/log_services.hpp

```diff
@@ -6719,6 +6719,103 @@ inline void getRedfishUriByDbusObjPath(
         std::array<const char*, 0>{});
 }
 
+/**
+ * @brief Populate the value of Message and MessageArgs[0]
+ *
+ * @param[i,o] entryJson  Appropriate entry json object in the Async response
+ * object
+ * @param[i]   guardType  The guardType string(Critical/Warning/Manual/Spare)
+ * @return void
+ */
+static void updateEntryMessageAndMessageArgs(nlohmann::json& entryJson,
+                                             const std::string& guardType)
+{
+    try
+    {
+        // Message format = Manual Gard Record Memory Module 1
+        // The pretty name at the end will be appended later.
+        entryJson["Message"] = guardType + " Gard Record ";
+        entryJson["MessageArgs"].push_back(guardType);
+    }
+    catch (const std::exception& e)
```

Comment:
Instead of adding a new element `guardType` to `MessageArgs` here,  can a new arg be added into `OpenBMC.0.6.GardRecord` (in openbmc.json file)?

Reply:
> Does this require the exception handling (`try-catch`)? Is it because json push_back may cause the exception? What condition would it be? Can we avoid it?

done